zio-http/src/main/scala/zio/http/codec/SegmentCodec.scala

```diff
@@ -0,0 +1,217 @@
+      if (index < 0 || index >= segments.length) -1
+      else {
+        val segment = segments(index)
+
+        if (segment == "true" || segment == "false") 1 else -1
+      }
+  }
+  private[http] final case class IntSeg(name: String, doc: Doc = Doc.empty)   extends SegmentCodec[Int]            {
+    def ??(doc: Doc): IntSeg = copy(doc = this.doc + doc)
+
+    def format(value: Int): Path = Path(s"/$value")
+
+    def matches(segments: Chunk[String], index: Int): Int = {
+      if (index < 0 || index >= segments.length) -1
+      else {
+        val SegmentCodec = segments(index)
+        var i            = 0
+        var defined      = true
+        while (i < SegmentCodec.length) {
+          if (!SegmentCodec.charAt(i).isDigit) {
+            defined = false
+            i = SegmentCodec.length
+          }
+          i += 1
+        }
+        if (defined && i >= 1) 1 else -1
+      }
+    }
+  }
+  private[http] final case class LongSeg(name: String, doc: Doc = Doc.empty)  extends SegmentCodec[Long]           {
+    def ??(doc: Doc): LongSeg = copy(doc = this.doc + doc)
+
+    def format(value: Long): Path = Path(s"/$value")
+
+    def matches(segments: Chunk[String], index: Int): Int = {
+      if (index < 0 || index >= segments.length) -1
+      else {
+        val SegmentCodec = segments(index)
+        var i            = 0
+        var defined      = true
+        while (i < SegmentCodec.length) {
+          if (!SegmentCodec.charAt(i).isDigit) {
+            defined = false
+            i = SegmentCodec.length
+          }
+          i += 1
+        }
+        if (defined && i >= 1) 1 else -1
+      }
+    }
+  }
+  private[http] final case class Text(name: String, doc: Doc = Doc.empty)     extends SegmentCodec[String]         {
+    def ??(doc: Doc): Text = copy(doc = this.doc + doc)
+
+    def format(value: String): Path = Path(s"/$value")
+
+    def matches(segments: Chunk[String], index: Int): Int =
+      if (index < 0 || index >= segments.length) -1
+      else 1
+  }
+  private[http] final case class UUID(name: String, doc: Doc = Doc.empty)     extends SegmentCodec[java.util.UUID] {
+    def ??(doc: Doc): UUID = copy(doc = this.doc + doc)
+
+    def format(value: java.util.UUID): Path = Path(s"/$value")
+
+    def matches(segments: Chunk[String], index: Int): Int = {
+      if (index < 0 || index >= segments.length) -1
+      else {
+        val SegmentCodec = segments(index)
+
+        var i       = 0
+        var defined = true
+        var group   = 0
+        var count   = 0
+        while (i < SegmentCodec.length) {
+          val char = SegmentCodec.charAt(i)
+          if ((char >= 48 && char <= 57) || (char >= 65 && char <= 70) || (char >= 97 && char <= 102))
+            count += 1
+          else if (char == 45) {
+            if (
+              group > 4 || (group == 0 && count != 8) || ((group == 1 || group == 2 || group == 3) && count != 4) || (group == 4 && count != 12)
```

Comment:
what kind of magic is this? :D

Reply:
Doesn't look right, does it. I hope I copy/pasted it from the old TextCodec but it looks more like Copilot nonsense.